third_party/nvfuser/csrc/codegen.cpp

```diff
@@ -492,8 +492,10 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     TORCH_INTERNAL_ASSERT(false, "Unreachable");
   }
 
-  void handle(const TensorView*) final {
-    TORCH_INTERNAL_ASSERT(false, "Unreachable");
+  void handle(const TensorView* tv) final {
+    // This allows us to access scalar tensor as if they are just scalar
+    TORCH_INTERNAL_ASSERT(tv->isZeroDim(), "TensorView can only be handled as scalar tensor");
+    code_ << ir_utils::varName(tv) << "[0]";
```

Comment:
This I think should be fine for scalar tensor. Seeking advises @naoyam @zasdfgbnm

Reply:
Instead of handling scalar tensor here, should we handle it during index lowering? In https://github.com/csarofeen/pytorch/blob/9340f80eab9393ea567fe18ad370bd6f31b8e062/third_party/nvfuser/csrc/ir_internal_nodes.h#L74-L76
We can check if `input(1)` is a tensor, if yes, create a `kir::TensorIndex` with `index` zero value.

Reply:
It should be fine, but it doesn't seem quite right to me. I think all we need to do is to lower the index input in `lower_index.cpp`. 

I'll work on cleaning it up. Can you add a test?

Reply:
Sure. Let me add a test on the python API~~ :bow: 
Thanks for following up on this one.

Reply:
Pushed a commit. The Python test seems fine. Running the other tests.